### ISSUE TITLE
Allow compiled V8 binary as application entry-point.

### ISSUE THREAD

**Issue by Kruithne** [nwjs/nw.js#7220]
To protect sources, they are compiled with V8 and then executed using `evalNWBin` on a Window. In order to execute this however, you need to specify either a HTML or JS entry-point in `package.json` which then loads the compiled binary.

It would be beneficial to reduce the needed route here if we could simply specify a compiled V8 binary as the entry-point.

```{
	"main": "source.bin"
}```

I am not familiar with the internals of nwjs, so I am unsure if a Window context is specifically required before hand. If this is the case, then simply spawning a blank window without any HTML using `window` sub-fields defined in the `package.json` should be fine, which the user can then obtain with `nw.Window.get()` or such.

Overall, it would be nice to reduce the need for having to include a `index.html` which just contains `<script>nw.Window.get().evalNWBin(frame, 'binary.bin');</script>` when all bootstrapping is done in the compiled script.

**Comment by Sombian** [comment]
Tis quite an extra essential.

**Comment by rogerwang** [comment]
It would be the same as pointing `main` to a JS file with a single line of calling `evalNWBin`. So it could be done in the application...

**Comment by NbrTDB** [comment]
I am very interested in this idea from a security/source code protection perspective, except I'm not sure it adds any extra of either... It's more convenient than having to have JS just to load your binaries, sure, but the user can just edit package.json and start executing their own code anyway.

Perhaps it would be useful if there were some way NW.js could be made to only use a binary as an entry point, such that it couldn't be changed... but that is starting to sound like a custom build to me, with hard-coded entry point and thus ignoring certain package.json parameters. 

Besides it still doesn't stop a user from just swapping out the binary with their own and using it as a wrapper... or swapping out a different version of NW.js that had no such restriction. Then they're back to being able to do things like open dev tools and ferret around for objects and functions present in the compiled source, or whatever other mischievous stuff they're up to.

**Comment by Kruithne** [comment]
I agree with you, there's not much real security gain to be had from this feature. Anyone who is invested in the security of their application will be compiling their own build anyway with more sophisticated measures.

I would mostly like to see this feature just to remove the need to have a JS file acting as nothing more than a bootstrap for the application. It would feel cleaner and more professional, I think?

At the end of the day, it may give some peace of mind to people who are not wanting to compile their own build, but also are slightly security conscious (even if it is generally just security via obscurity and only a small hurdle for people to get over).

**Comment by NbrTDB** [comment]
Yeah it would be much tidier to be able to specify a binary directly in package.json. I feel like it's worthwhile just for the convenience, if nothing else.

**Comment by rudasn** [comment]
@Kruithne 

> At the end of the day, it may give some peace of mind to people who are not wanting to compile their own build, but also are slightly security conscious (even if it is generally just security via obscurity and only a small hurdle for people to get over).

Is there a better way to make our own builds (compiled .bin) to make sure that the code stays protected?

Considering the current build system (`nwjc`), are there any tools that someone can use to get a look at the contained source code? Or rather, Is it possible to ship an nwjs compiled binary that contains a piece of text (string) and be sure that it will not be leaked?

Thanks
